cr17/nodes/cr17/high_level_state_controller.py

```diff
@@ -230,31 +235,47 @@ def run(self):
             if(self.autostate == "INIT"):
                 if(True):#change to some actual check
                     self.autostate = "F_OBSTACLE_FIELD"
+
+                    msg = Bool()
+                    msg.data = False
+                    self.puber.publish(msg)
+                    rospy.logwarn("Publishing out path desire.")
             elif(self.autostate == "F_OBSTACLE_FIELD"):
                 if(self.pose.position.y >= Y_IN_MINING_AREA):
                     self.autostate = "MINING_BEHAVIOR"
             elif(self.autostate == "MINING_BEHAVIOR"):
                 if(not self.autonomyEnabled or (self.pose.position.y < Y_IN_MINING_AREA - .3)):
                     self.autostate = "B_OBSTACLE_FIELD"
                     self.minePath = None
+
+                    msg = Bool()
+                    msg.data = True
+                    self.puber.publish(msg)
+                    rospy.logwarn("Publishing in path desire.")
+
             elif(self.autostate == "B_OBSTACLE_FIELD"):
                 if(self.pose.position.y <= Y_IN_DOCKING_AREA):
                     self.autostate = "DOCKING"
             elif(self.autostate == "DOCKING"):
                 if(self.pose.position.y < Y_IN_DUMP_RANGE):
                     self.autostate = "DUMPING"
             elif(self.autostate == "DUMPING"):
-                if(dumping_complete or (self.pose.position.y >= Y_IN_DUMP_RANGE+0.5)):
+                if(dumping_complete or (self.pose.position.y >= Y_IN_DUMP_RANGE)):
                     self.autostate = "F_OBSTACLE_FIELD"
                     self.dumpTimer = None
+
+                    msg = Bool()
+                    msg.data = False
+                    self.puber.publish(msg)
+                    rospy.logwarn("Publshing out path desire.")
```

Comment:
Where is it publishing the out path desire? There should be no paths published in the if statements that are meant for state changes.